src/jquery.interactive-world-map.js

```diff
@@ -1,142 +1,180 @@
-                }
-                var $this = $(this);
-                $element   = $('g', $this).find("[data-code='" + code  + "']");
-                $offset    = $element.offset();
-
-                $(".main-modal, #modal-background").toggleClass("active");
-                // Calculate the distance between the country and the top for show with the correctly space the popup
-                distanceXtop = $offset.top  - marginCountryPopupForTop;
-                distanceLeft = $offset.left + marginCountryPopupForLeft;
-
-                var first           = dataCountries[code];
-                countryName         = first['name'];
-                description         = first['description'];
-                mainImage           = first['main_image'];
-
-                $(".main-modal" ).css("top", distanceXtop + 'px').css("left", distanceLeft + 'px');
-                $(".main-modal .description").text(description);
-                $(".main-modal .title").text(countryName);
-                $(".main-image").attr('src', mainImage);
+    function setPatternAttributes(x, y, cx, cy, r) {
+        _x  = x;
+        _y  = y;
+        _cx = cx;
+        _cy = cy;
+        _r  = r;
+    }
 
-            }
+    function createPatternSvg(id, color) {
+        return '<pattern id="' + id + '" width="5" height="5"     patternUnits="userSpaceOnUse"><circle x="' + _x + '" y="' + _y + '" cx="' + _cx + '" cy="' + _cy + '" r="' + _r + '" style="stroke:none; fill:' + color + '"></circle></pattern>';
+    }
 
-        });
-        //Configurations for the map. Dotted Style and popup prepended
-        colorHighlighted = $.fn.interactiveWorldMap.defaults.highlighted;
-        colorHover       = $.fn.interactiveWorldMap.defaults.hover;
-        pattern  = '<pattern id="dotsGeneral" width="5" height="5" patternUnits="userSpaceOnUse"><circle x="1" y="1" cx="3" cy="3" r="3" style="stroke:none; fill:#cccccc;"></circle></pattern>';
-        pattern2 = '<pattern id="dotsHighlighted" width="5" height="5" patternUnits="userSpaceOnUse"><circle x="1" y="1" cx="3" cy="3" r="3" style="stroke:none; fill:' + colorHighlighted + '"></circle></pattern>';
-        patternHover = '<pattern id="dotsHover" width="5" height="5" patternUnits="userSpaceOnUse"><circle x="1" y="1" cx="3" cy="3" r="3" style="stroke:none; fill:' + colorHover+ '"></circle></pattern>';
-        $(this).parent().prepend('<div class="main-modal"><div class="container"><div class="title"></div><div class="container-image" ><img src="" class="main-image" alt=""></div><p><h2></h2><div class="description"></div></p></div></div>');
-        $('defs').html(pattern+pattern2+patternHover);
-    };
+    function prependModalContainer(element) {
+        $(element).parent().prepend('<div class="main-modal"><div class="container"><div class="title"></div><div class="container-image" ><img src="" class="main-image" alt=""></div><p><h2></h2><div class="description"></div></p></div></div>');
+    }
```

Comment:
Because the element I'm working is the .world-map and the popup I was working in it outside of the .world-map container that's why.